### DISCUSSION TITLE
Feature Request: Classes

### DISCUSSION THREAD

**Discussion by saschabregenhorn** [gearsdigital/enhanced-toolbar-link-dialog#85] [General]
It would be super nice to have predefined classes:

`<a class="button" href=""></a>`

![classes](https://user-images.githubusercontent.com/2510090/187886872-8789f49c-580a-465c-9b2a-f33c9cf35b2b.png)

**Comment by gearsdigital** [comment]
Thank you for reaching out @saschabregenhorn... Your idea is awesome and would be great addition to the existing feature set! Sadly (and I regret I've to say that) it isn't (easily) possible right now. The simple reason is: I've no control over the generated markup.

Let me explain... the HTML Markup is generated [here](https://github.com/getkirby/kirby/blob/3.7.5/panel/src/components/Forms/Writer/Marks/Link.js#L75). Afaik it's not possible to override or extend the above mentioned `Link`-class because it's compiled into the the panel and there is no hook or something which I can use to alter the generated markup by adding attributes other then defined.

A possible solution might be to disable the default Link-Mark (Icon) and create a custom one (a similar approach was [discussed here](https://github.com/gearsdigital/enhanced-toolbar-link-dialog/issues/74#issuecomment-1216445471)). A custom Mark would provide full control about attributes and everything. It would also decouple the plugin from the current core dialog but I've currently no spare time to build that 🥶

I'll leave your request open, maybe I (or someone else) can find some time and implement that.

**Comment by saschabregenhorn** [comment]
> I'll leave your request open, maybe I (or someone else) can find some time and implement that.

All right, thanks for the explanation! 🤘🏻
